flocker/common/_retry.py

```diff
@@ -112,29 +113,42 @@ def loop(result):
     return d.addActionFinish()
 
 
-def timeout(reactor, deferred, timeout_sec):
-    """Adds a timeout to an existing deferred. If the timeout expires before
-    the deferred expires, then the deferred is cancelled.
+def timeout(reactor, deferred, timeout_sec, reason=None):
```

Comment:
:question: `timeout_sec` should probably be a `timedelta`. Can wait for a different branch though.